packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,239 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as path from "path";
+import { readdirSync } from "fs";
+import { strict as assert } from "assert";
+import {
+	createWeightedAsyncGenerator as createWeightedGenerator,
+	AsyncGenerator as Generator,
+	takeAsync as take,
+} from "@fluid-internal/stochastic-test-utils";
+import {
+	createDDSFuzzSuite,
+	DDSFuzzModel,
+	DDSFuzzHarnessEvents,
+	DDSFuzzSuiteOptions,
+} from "@fluid-internal/test-dds-utils";
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import {
+	IFluidDataStoreRuntime,
+	IChannelServices,
+	IChannelAttributes,
+} from "@fluidframework/datastore-definitions";
+import { SharedStringFactory } from "../sequenceFactory";
+import {
+	appendAddIntervalToRevertibles,
+	// appendChangeIntervalToRevertibles,
+	// appendDeleteIntervalToRevertibles,
+	// appendIntervalPropertyChangedToRevertibles,
+	// appendSharedStringDeltaToRevertibles,
+} from "../revertibles";
+import { SharedString } from "../sharedString";
+import { assertEquivalentSharedStrings } from "./intervalUtils";
+import {
+	Operation,
+	FuzzTestState,
+	RevertOperation,
+	makeReducer,
+	RevertibleSharedString,
+	isRevertibleSharedString,
+} from "./intervalCollection.fuzzUtils";
+import { makeOperationGenerator } from "./intervalCollection.fuzz.spec";
+import { minimizeTestFromFailureFile } from "./intervalCollection.fuzzMinimization";
+
+// Note: none of these options are currently exercised, since the fuzz test fails with pretty much
+// any configuration due to known bugs. Once shared interval collections are in a better state these
+// should be revisited.
+interface OperationGenerationConfig {
+	/**
+	 * Maximum length of the SharedString (locally) before no further AddText operations are generated.
+	 * Note due to concurency, during test execution the actual length of the string may exceed this.
+	 */
+	maxStringLength?: number;
+	/**
+	 * Maximum number of intervals (locally) before no further AddInterval operations are generated.
+	 * Note due to concurency, during test execution the actual number of intervals may exceed this.
+	 */
+	maxIntervals?: number;
+	maxInsertLength?: number;
+	intervalCollectionNamePool?: string[];
+	propertyNamePool?: string[];
+	validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+	maxStringLength: 1000,
+	maxIntervals: 100,
+	maxInsertLength: 10,
+	intervalCollectionNamePool: ["comments"],
+	propertyNamePool: ["prop1", "prop2", "prop3"],
+	validateInterval: 100,
+};
+
+// Since the clients are created by the fuzz harness, the factory object must be
+// modified in order to set the mergeTreeUseNewLengthCalculations option on the
+// underlying merge tree.
+class RevertibleFactory extends SharedStringFactory {
```

Comment:
i think actually with #15868 merged, you can get rid of this now.

Reply:
nice case of concurrent changes simplifying things :)

Reply:
Is it a problem that the above pr went into next, not main?